subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -162,13 +165,23 @@ VirtualFileSystem createVirtualFileSystem(
                 listenerManager.addListener(new RootBuildLifecycleListener() {
                     @Override
                     public void afterStart(GradleInternal gradle) {
-                        Map<String, String> systemPropertiesArgs = gradle.getStartParameter().getSystemPropertiesArgs();
+                        StartParameter startParameter = gradle.getStartParameter();
+                        Map<String, String> systemPropertiesArgs = startParameter.getSystemPropertiesArgs();
                         if (isRetentionEnabled(systemPropertiesArgs)) {
-                            Iterable<String> changedPathsSinceLastBuild = getChangedPathsSinceLastBuild(systemPropertiesArgs);
-                            for (String changedPathSinceLastBuild : changedPathsSinceLastBuild) {
+                            FileResolver fileResolver = new BaseDirFileResolver(startParameter.getCurrentDir(), () -> {
```

Comment:
I probably wouldn't have used a `FileResolver` here, but a simple `File`. Then the relative paths would be converted into the absolute paths by using `new File(currentDir, relativePath).getAbsolutePath()`.

Why do you want to use a `FileResolver` here?

Reply:
Because we might want to pass in absolute files as well, and then I'd have had to add a check whether the path is absolute. And we already have that logic in the `FileResolver`.

All this code will come out as soon as we have some sort of file watching. So even if this is not the most optimal I think it's fine.

Reply:
Why do you need a check for absolute paths? `new File(baseDir, absolutePath)` already returns the correct result, aka pointing to the `absolutePath`, doesn't it?

Reply:
It doesn't.